user_guide_src/source/extending/composer_packages.rst

```diff
@@ -0,0 +1,195 @@
+##########################
+Creating Composer Packages
+##########################
+
+You can make the :doc:`../general/modules` you create into Composer packages,
+or can create a Composer package for CodeIgniter 4.
+
+.. contents::
+    :local:
+    :depth: 2
+
+****************
+Folder Structure
+****************
+
+Here's a typical directory structure for a Composer package::
+
+    your-package-name/
+    ├── .gitattributes
+    ├── .gitignore
+    ├── LICENSE
+    ├── README.md
+    ├── composer.json
+    ├── src/
+    │   └── YourClass.php
+    └── tests/
+        └── YourClassTest.php
+
+**********************
+Creating composer.json
+**********************
+
+In the root of your package directory, create a **composer.json** file. This file
+defines metadata about your package and its dependencies.
+
+The ``composer init`` command helps you create it.
+
+For example, **composer.json** might look like this::
+
+    {
+        "name": "your-vendor/your-package",
+        "description": "Your package description",
+        "type": "library",
+        "license": "MIT",
+        "autoload": {
+            "psr-4": {
+                "YourVendor\\YourPackage\\": "src/"
+            }
+        },
+        "authors": [
+            {
+                "name": "Your Name",
+                "email": "yourname@example.com"
+            }
+        ],
+        "require": {
+            // Any dependencies required by your package go here
+        },
+        "require-dev": {
+            // Any development dependencies (e.g., PHPUnit) go here
+        }
+    }
+
+Package Name
+============
+
+The ``name`` field is important here. Package names are generally written in the
+format "vendor-name/package-name" with all lowercase. Here is a common example:
+
+- ``your-vendor-name``: The name that identifies the vendor (creator of the package),
+  such as the name of you or your organization.
+- ``your-package-name``: The name of the package you are creating.
+
+Thus, it is important to make the name unique and distinguish it from other packages.
+Uniqueness is especially important when publishing.
+
+Namespace
+=========
+
+The package name then determines the vendor namespace in ``autoload.psr4``.
+
+If your package name is ``your-vendor/your-package``, the vendor namespace must
```

Comment:
Not always true. I see packages where it is all lowercase. Some removes some letters.
```suggestion
If your package name is ``your-vendor/your-package``, the vendor namespace will typically
```

Reply:
> I see packages where it is all lowercase. 

I see. It could be. PHP's classnames are case insensitive.

> Some removes some letters.

Why? Isn't it just a incorrect use, or just a typo?
If it is not the same, the vendor namespace might conflict.

Reply:
like in `pestphp/pest`, the vendor namespace is Pest.

Reply:
Isn't it just a incorrect use?
If someone gets the vendor name `pest` on packagist, they would use the vendor name space `Pest`?

Reply:
If we follow the PSR, the class name is CamelCase. So I think the current wording is fine.
Or not `must` but `should`?

Reply:
Let's leave it as is for now.